sardana_limaccd/client.py

```diff
@@ -0,0 +1,323 @@
+        if acq_status not in {"Ready", "Running"}:
+            return acq_status
+        if self.stopped:
+            return "Ready"
+        done = idx_saved if self.lima.saving.enabled else idx_ready
+        if done < self.config["acq_nb_frames"] - 1:
+            acq_status = "Running"
+        trig_mode = self.config["acq_trigger_mode"]
+        if ready_for_next and self.is_int_trig_multi() and acq_status == "Running":
+            if (idx_ready + 1) >= self.nb_starts:
+                acq_status = "Ready"
+        return acq_status
+
+    def next_frame(self):
+        last = self['last_image_ready']
+        n = self._acq_next_number
+        if n > last:
+            # no frame available yet
+            return
+        frame = self.lima.read_frames(n, n + 1)[0]
+        self._acq_next_number += 1
+        return frame
+
+    def next_frames(self):
+        last = self['last_image_ready']
+        start = self._acq_next_number
+        if start > last:
+            # no frame available yet
+            return []
+        frames = self.lima.read_frames(start, last + 1)
+        self._acq_next_number += len(frames)
+        return frames
+
+    def next_ref_frame(self):
+        last = self["last_image_saved"]
+        n = self._save_next_number + last
+        return self.lima.saving.filename(n)
+
+    def next_ref_frames(self):
+        last = self["last_image_saved"]
+        n = last - self._last_saved_number
+        refs = [
+            self.lima.saving.filename(self._save_next_number + i)
+            for i in range(n)
+        ]
+        self._save_next_number += n
+        self._last_saved_number += n
+        return refs
+
+
+class Saving(object):
+
+    FILE_PATTERN = \
+        "{scheme}://{saving_directory}/{saving_prefix}{index}{saving_suffix}"
+
+    def __init__(self, lima):
+        self.lima = lima
+        self.first_image_nb = 0
+        self.enabled = False
+        self.pattern = ""
+        self.config = {}
+
+    def filename(self, index):
+        scheme = "file"
+        if self.config["saving_format"] == "HDF5":
+            scheme = "h5file"
+        index = self.config["saving_index_format"] % index
+        return self.FILE_PATTERN.format(
+            scheme=scheme, index=index, **self.config
+        )
+
+    def prepare(self):
+        if not self.enabled:
+            self.config = {"saving_mode": "MANUAL"}
+            self.lima["saving_mode"] = "MANUAL"
+            return
+        self.config = config = saving_for_pattern(self.pattern)
+        config["saving_mode"] = "AUTO_FRAME"
+        config["saving_overwrite_policy"] = "ABORT"
+        names, values = zip(*config.items())
+        self.lima[names] = values
+        if self.first_image_nb != 0:
+            self.lima["saving_next_number"] = -1
+            time.sleep(0.05) # empirical value?
+            self.lima["saving_prefix"] = config["saving_prefix"]
+            t0 = time.monotonic()
```

Comment:
`time.monotonic` is not available for python < 3.3. Some third party modules implement this method for old python version (https://pypi.org/project/monotonic/) but maybe is better to do not add an extra dependency if can be easily avoided.

Reply:
Yes, you are right.

I guess we can do something simple like:
```python
 monotonic = getattr(time, "monotonic", time.time)
```

Reply:
Indeed, once ready I can test the code again.